File: iphone-code/code/Stasher/STBearPopUpViewController.m
//
//  STBearPopUpViewController.m
//  Stasher
//
//  Created by Bhushan on 25/03/15.
//  Copyright (c) 2015 OAB. All rights reserved.
//

#import "STBearPopUpViewController.h"

@interface STBearPopUpViewController ()

@end

@implementation STBearPopUpViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view from its nib.
    self.rectangularView.clipsToBounds = YES;
    self.rectangularView.layer.cornerRadius = 10.0f;
    self.rectangularView.layer.borderColor=[UIColor clearColor].CGColor;
    self.rectangularView.layer.borderWidth=2.0f;
    [self.buttonPopUp.titleLabel setFont:[UIFont FontForButtonsWithSize:11.0f]];
    [self.buttonLater.titleLabel setFont:[UIFont FontForButtonsWithSize:11.0f]];
    [self.buttonSetUp.titleLabel setFont:[UIFont FontForButtonsWithSize:11.0f]];
    
    [self.labelPopUpText setFont:[UIFont FontGothamRoundedBookWithSize:11.0f]];
    [self.labelPopUpText setTextColor:[UIColor stasherTextColor]];
    NSArray *imageNames = @[@"Bear1.png", @"Bear2.png", @"Bear3.png", @"Bear4.png",
                            @"Bear5.png", @"Bear6.png", @"Bear7.png", @"Bear8.png",
                            @"Bear9.png", @"Bear10.png"];
    
    NSMutableArray *images = [[NSMutableArray alloc] init];
    for (int i = 0; i < imageNames.count; i++) {
        [images addObject:[UIImage imageNamed:[imageNames objectAtIndex:i]]];
    }
    self.imgViewBear.animationImages = images;
    self.imgViewBear.animationDuration = kBearAnimationDuration;
    [self.imgViewBear startAnimating];
    [self.rectangularView setBackgroundColor:[UIColor stasherPopUpBGColor]];
    [self.buttonPopUp setHidden:NO];
    [self.buttonLater setHidden:YES];
    [self.buttonSetUp setHidden:YES];
    if (self.BearPopUpKind == ADDPARENTBEARPOPUP) {
        [self.buttonPopUp setTitle:@"ADD PARENT" forState:UIControlStateNormal];
        [self.labelPopUpText setText:@"To use Stasher, You need to add a parent."];
    }else if (self.BearPopUpKind == THANKSFORADDINGPARENTBEARPOPUP){
        [self.buttonPopUp setTitle:@"Done, Let’s go" forState:UIControlStateNormal];
        [self.labelPopUpText setText:@"Thanks for adding a parent!"];
    }else if (self.BearPopUpKind == ADDCHILDBEARPOPUP){
        [self.buttonPopUp setTitle:@"ADD CHILD" forState:UIControlStateNormal];
        [self.labelPopUpText setText:@"Let's start by addding a child."];
    }else if (self.BearPopUpKind == THANKSFORADDINGACHILDBEARPOPUP){
        [self.buttonPopUp setHidden:YES];
        [self.buttonLater setHidden:NO];
        [self.buttonSetUp setHidden:NO];
        [self.labelPopUpText setText:@"Thanks for adding a child, to use Stasher you have to set up your Bank Details."];
    }else if (self.BearPopUpKind == BANKACCOUNTLATERBEARPOPUP){
        [self.buttonPopUp setTitle:@"OK" forState:UIControlStateNormal];
        [self.labelPopUpText setText:@"No that's OK, we can do it later."];
    }else if (self.BearPopUpKind == BANKACCOUNTDONEBEARPOPUP){
        [self.buttonPopUp setTitle:@"OK" forState:UIControlStateNormal];
        [self.labelPopUpText setText:@"Cool, we never share this information to anyone."];
    }else if (self.BearPopUpKind == THANKSFORADDINGACHILDNOBANKBEARPOPUP){
        [self.buttonPopUp setTitle:@"OK" forState:UIControlStateNormal];
        [self.labelPopUpText setText:@"Thanks for adding a child."];
    }
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

- (IBAction)popUpGreenButtonPressed:(id)sender
{
    if ([self.delegate respondsToSelector:@selector(bearPopUpButtonPressedWithPopUpKind:)]) {
        [self.delegate bearPopUpButtonPressedWithPopUpKind:self.BearPopUpKind];
    }
}

- (IBAction)laterButtonPressed:(id)sender
{
    if ([self.delegate respondsToSelector:@selector(bearPopUpLaterButtonPressed)]) {
        [self.delegate bearPopUpLaterButtonPressed];
    }
}

- (IBAction)setUpButtonPressed:(id)sender
{
    if ([self.delegate respondsToSelector:@selector(bearPopUpSetUpButtonPressed)]) {
        [self.delegate bearPopUpSetUpButtonPressed];
    }
}


@end

File: swift-code/Stasher/STBearPopUpViewController.swift
//  Converted with Swiftify v1.0.6292 - https://objectivec2swift.com/
//
//  STBearPopUpViewController.swift
//  Stasher
//
//  Created by Bhushan on 25/03/15.
//  Copyright (c) 2015 OAB. All rights reserved.
//
import UIKit
protocol BearPopUpDelegate: NSObjectProtocol {
    func bearPopUpButtonPressed(withPopUpKind bearPopUpKind: BearPopUpType)

    func bearPopUpLaterButtonPressed()

    func bearPopUpSetUpButtonPressed()
}
class STBearPopUpViewController: UIViewController {

    weak var delegate: BearPopUpDelegate?
    @IBOutlet var imgViewTransparentBG: UIImageView!
    @IBOutlet var imgViewBear: UIImageView!
    @IBOutlet var rectangularView: UIView!
    @IBOutlet var labelPopUpText: UILabel!
    @IBOutlet var buttonPopUp: UIButton!
    @IBOutlet var buttonLater: UIButton!
    @IBOutlet var buttonSetUp: UIButton!
    var bearPopUpKind = BearPopUpType()


    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view from its nib.
        rectangularView.clipsToBounds = true
        rectangularView.layer.cornerRadius = 10.0
        rectangularView.layer.borderColor = UIColor.clear.cgColor
        rectangularView.layer.borderWidth = 2.0
        buttonPopUp.titleLabel?.font = UIFont(size: 11.0)
        buttonLater.titleLabel?.font = UIFont(size: 11.0)
        buttonSetUp.titleLabel?.font = UIFont(size: 11.0)
        labelPopUpText.font = UIFont.fontGothamRoundedBook(withSize: 11.0)
        labelPopUpText.textColor = UIColor.stasherText()
        var imageNames: [Any] = ["Bear1.png", "Bear2.png", "Bear3.png", "Bear4.png", "Bear5.png", "Bear6.png", "Bear7.png", "Bear8.png", "Bear9.png", "Bear10.png"]
        var images = [Any]()
        for i in 0..<imageNames.count {
            images.append(UIImage(named: imageNames[i]))
        }
        imgViewBear.animationImages = images
        imgViewBear.animationDuration = kBearAnimationDuration
        imgViewBear.startAnimating()
        rectangularView.backgroundColor = UIColor.stasherPopUpBG()
        buttonPopUp.isHidden = false
        buttonLater.isHidden = true
        buttonSetUp.isHidden = true
        if bearPopUpKind == ADDPARENTBEARPOPUP {
            buttonPopUp.setTitle("ADD PARENT", for: .normal)
            labelPopUpText.text = "To use Stasher, You need to add a parent."
        }
        else if bearPopUpKind == THANKSFORADDINGPARENTBEARPOPUP {
            buttonPopUp.setTitle("Done, Let’s go", for: .normal)
            labelPopUpText.text = "Thanks for adding a parent!"
        }
        else if bearPopUpKind == ADDCHILDBEARPOPUP {
            buttonPopUp.setTitle("ADD CHILD", for: .normal)
            labelPopUpText.text = "Let's start by addding a child."
        }
        else if bearPopUpKind == THANKSFORADDINGACHILDBEARPOPUP {
            buttonPopUp.isHidden = true
            buttonLater.isHidden = false
            buttonSetUp.isHidden = false
            labelPopUpText.text = "Thanks for adding a child, to use Stasher you have to set up your Bank Details."
        }
        else if bearPopUpKind == BANKACCOUNTLATERBEARPOPUP {
            buttonPopUp.setTitle("OK", for: .normal)
            labelPopUpText.text = "No that's OK, we can do it later."
        }
        else if bearPopUpKind == BANKACCOUNTDONEBEARPOPUP {
            buttonPopUp.setTitle("OK", for: .normal)
            labelPopUpText.text = "Cool, we never share this information to anyone."
        }
        else if bearPopUpKind == THANKSFORADDINGACHILDNOBANKBEARPOPUP {
            buttonPopUp.setTitle("OK", for: .normal)
            labelPopUpText.text = "Thanks for adding a child."
        }

    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    /*
    #pragma mark - Navigation
    
    // In a storyboard-based application, you will often want to do a little preparation before navigation
    - (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
        // Get the new view controller using [segue destinationViewController].
        // Pass the selected object to the new view controller.
    }
    */

    @IBAction func popUpGreenButtonPressed(_ sender: Any) {
        if delegate?.responds(to: #selector(self.bearPopUpButtonPressedWithPopUpKind)) {
            delegate?.bearPopUpButtonPressed(withPopUpKind: bearPopUpKind)
        }
    }

    @IBAction func laterButtonPressed(_ sender: Any) {
        if delegate?.responds(to: #selector(self.bearPopUpLaterButtonPressed)) {
            delegate?.bearPopUpLaterButtonPressed()
        }
    }

    @IBAction func setUpButtonPressed(_ sender: Any) {
        if delegate?.responds(to: #selector(self.bearPopUpSetUpButtonPressed)) {
            delegate?.bearPopUpSetUpButtonPressed()
        }
    }
}